yolov3_tf2/dataset.py

```diff
@@ -267,23 +266,107 @@ def parse_xml(xml):
     return {xml.tag: result}
 
 
-def parse_set(class_map, out_file, annotations_dir, images_dir):
+def parse_set(class_map, out_file, annotations_dir, images_dir, aumentation):
     writer = tf.io.TFRecordWriter(out_file)
 
     for annotation_file in tqdm.tqdm(os.listdir(annotations_dir)):
         if not annotation_file.endswith('.xml'):
             continue
 
-        # print("file: " + annotation_file)
         annotation_xml = os.path.join(annotations_dir, annotation_file)
         annotation_xml = lxml.etree.fromstring(open(annotation_xml).read())
         annotation = parse_xml(annotation_xml)['annotation']
 
+        height, width = get_image_dimensions(annotation, images_dir)
+        try:
+            xml_data_dict = extract_xml_data(annotation, class_map, height, width)
+        except Exception:
+            continue
+
+        if len(xml_data_dict['classes']) > 100:
+            logging.error(f"too many classes ({len(xml_data_dict['classes'])}) on {annotation_xml}")
+            continue
+
         try:
-            tf_example = build_example(annotation, class_map, images_dir)
-            writer.write(tf_example.SerializeToString())
+            raw_image, key = open_image(annotation, images_dir, xml_data_dict)
         except:
-            pass
+            continue
+
+        tf_example = build_example(annotation['filename'], images_dir, xml_data_dict, raw_image, key)
+        writer.write(tf_example.SerializeToString())
+
+        if aumentation:
+            tf_examples = augment_image(annotation, images_dir, xml_data_dict, class_map)
+            for tf_example in tf_examples:
+                writer.write(tf_example.SerializeToString())
 
     writer.close()
     logging.info(f"Wrote {out_file}")
+
+
+def get_image_dimensions(annotation, images_dir):
+
+    img_path = os.path.join(
+        images_dir, annotation['filename'].replace('set_01', '').replace(".xml", ".jpg")
+    )
+
+    try:
+        width = int(annotation['size']['width'])
+        height = int(annotation['size']['height'])
+    except KeyError:
+        im = Image.open(img_path)
+        width, height = im.size
+    width = width if width > 0 else 416
+    height = height if height > 0 else 416
+
+    return height, width
+
+
+def extract_xml_data(annotation, class_map, height, width) -> DefaultDict:
```

Comment:
mude `extract` para `parse` e adicione o formato ao nome da função, `parse_xml_data` pode ser qualquer coisa (data é uma palavra especialmente ruim por ser generiaca, precisamos especificar os nossos dados). por exemplo, se for o formato do dataset Pascal VOC, poderiamos renomear a função para `parse_pascal_voc_xml`

Reply:
9b75fc0